.github/workflows/build.yml

```diff
@@ -0,0 +1,79 @@
+name: Build
+
+on: [ push, pull_request, workflow_dispatch ]
+
+jobs:
+  node:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up chromedriver
+        uses: nanasess/setup-chromedriver@master
+
+      - run: |
+         sudo apt-get install xvfb
+         export DISPLAY=:99
+         chromedriver --url-base=/wd/hub &
+         sudo Xvfb -ac :99 -screen 0 1280x1024x24 > /dev/null 2>&1 & # optional
+
+      - name: "Node Cache"
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-cache-node-${{ hashFiles('**/package.json', '**/package-lock.json') }}
+
+      - name: "Node Cache Modules"
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-cache-node_modules-${{ hashFiles('**/package.json', '**/package-lock.json') }}
+
+      - name: "Setup Node"
+        uses: actions/setup-node@v2
+        with:
+          node-version: 10
+
+      - name: "Install Node"
+        run: npm install
+
+      - name: "Node Tests"
+        run: |
+          npm run build
+          xvfb-run --auto-servernum npm run test:ci
+
+      - name: Reporting
+        run: |
+          xvfb-run --auto-servernum cat ${{ github.workspace }}/static/weaver-components/reports/coverage/wvr-elements/lcov.info | ${{ github.workspace }}/node_modules/coveralls/bin/coveralls.js
+          xvfb-run --auto-servernum npm run build:static-production
+
+      - name: "Send to Coveralls (build node-${{ github.run_number }})"
+        uses: coverallsapp/github-action@v1.1.2
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          coveralls-repo-token: ${{ secrets.COVERALLS_REPO_TOKEN }}
+          path-to-lcov: ${{ github.workspace }}/static/weaver-components/reports/coverage/wvr-elements/lcov.info
+
+      - name: cleanup xvfb
+        run: |
+          xvfb_pids=`ps aux | grep tmp/xvfb-run | grep -v grep | awk '{print $2}'`
+          if [ "$xvfb_pids" != "" ]; then
+            echo "Killing the following xvfb processes: $xvfb_pids"
+            sudo kill $xvfb_pids
+          else
+            echo "No xvfb processes to kill"
+          fi
+
+      - name: "Deploy GH-Pages"
```

Comment:
Shouldn't this be in a separate Workflow file (like ghpages.yml)?

Reply:
It could be, but that is still under debate. Should we have duplicated workflow 'code' for a badge that we haven't had before and currently do not expose.

Reply:
My understanding is that there is a button that this is to provide and a separate Workflow is required for a separate button.

The issue at hand, explicitly asks for a GHPages Badge.
(https://github.com/TAMULib/weaver-components/issues/295)

Reply:
The badge is created for every successful workflow, yes.

I think that checklist is more than just expected badges.  @jeremythuff  was it intended that we have a badge for deploy to gh pages? If so, what was the rationale? We never had a badge for deploy to gh pages and it serves little purpose as a status check will fail the push to notify gh pages deploy failed anyhow. I just don't see the benefit in maintaining a bunch of additional workflow yaml files that are like 90% duplication just to show a badge in the readme.

Reply:
@wwelling , @kaladay  - Thank you for reviewing , approving PR

Reply:
Update : all the URL's are functional - however the header and footer components are not rendered  -  [Main Page](https://tamulib.github.io/weaver-components/) due to config.json file missing.